packages/core/src/extensions/lens-extension.ts

```diff
@@ -93,7 +106,7 @@ export class LensExtension<
    * folder name.
    */
   async getExtensionFileFolder(): Promise<string> {
-    return this[lensExtensionDependencies].fileSystemProvisionerStore.requestDirectory(this.id);
+    return this[lensExtensionDependencies].fileSystemProvisionerStore.requestDirectory(this.dataLocation);
```

Comment:
The id seems to be absolute path: https://github.com/lensapp/lens/blob/master/packages/core/src/extensions/extension-discovery/extension-discovery.ts#L354

Therefore, if we set storeName to "foo-bar" and we rename "foo-bar" to "@corp/foo-bar", the data location will change.

Reply:
Maybe the fix is something like (pseudocode): `fileSystemProvisionerStore.requestDirectory(getExtensionId(this.storeName || this.name))`

Reply:
I'll check that out.

Reply:
It be probably be safer to have the `extension.id` be calculated by `storeName` if it exists. That way you don't need to remember to use `this.storeName || this.name` everywhere you need `extension.id`

Reply:
@jweak good point. The only 'drawback' is that it's slightly surprising that `storeName` affects extension id.. but perhaps fine.

Reply:
Updated after @jweak's comment, provided I understood it correctly.

Reply:
I don't think it's right. So right now the LDK extension data is is `/Users/jheikka/Library/Application Support/Lens/extension_data/5d0651a6f92801496da1fe4fa20e1585ceeb4fa127282df2cf25d4f294e92a8d`. So we need to make sure that even after we update the name of the extension, directory name will still be `5d0651a6f92801496da1fe4fa20e1585ceeb4fa127282df2cf25d4f294e92a8d ` so that we don't lose data. So if it was calculated by `extension.name` we need to make sure it is now calculated by `storeName`

Reply:
I think the place is here: 

https://github.com/lensapp/lens/blob/f445bd4eb982bb6683407e20b29858ece97a9f49/packages/core/src/extensions/extension-loader/file-system-provisioner-store/file-system-provisioner-store.ts#L39-L60

Reply:
So maybe this is working after all

Reply:
I think the result is saved to `this.registeredExtensions` which is ultimately persisted to `lens-filesystem-provisioner-store.json` so the random bytes are not called again if it has already been calculated before.